packages/block-editor/src/store/actions.js

```diff
@@ -271,7 +275,7 @@ export const multiSelect = ( start, end ) => ( { select, dispatch } ) => {
 		return;
 	}
 
-	dispatch( { type: 'MULTI_SELECT', start, end } );
+	dispatch( { type: 'MULTI_SELECT', start, end, initialPosition } );
```

Comment:
Good idea to get wider input, thanks!

Because this hooks into the existing behaviour of the `initialPosition` state, we probably can't rename it / add an `__experimental` prefix. In this case, we're using the existing logic from initialPosition as used on an individual block in `selectBlock` and re-using the same logic for `multiSelect`, so it's more that the usage of it is experimental rather than that the state is experimental. Let me know if you can think of a good way for us to flag the usage, though!

Reply:
I'm mostly hesitant because I don't know `initialPosition` very well.

The `multiSelect` function signature is the public API here really, while the dispatch is an internal API. So I left my comment on the wrong line 😄 

If we want to make it experimental it should be ok to change line 266 to this:
```js
export const multiSelect = ( start, end, __experimentalInitialPosition = 0 ) => ( {
```

and then this line to:
```js
dispatch( { type: 'MULTI_SELECT', start, end, initialPosition: __experimentalInitialPosition } );
```

This way the `multiSelect` function's param is documented as experimental.

Reply:
Ah, perfect, thanks @talldan of course, that makes good sense now. I'll update the function signature 😀

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/38314/commits/db566abd56fb685edeabff48f3cbe437ece1db7b